src/coreneuron/io/nrn_filehandler.hpp

```diff
@@ -127,16 +127,13 @@ class FileHandler {
         mapinfo->name = std::string(name);
 
         if (nseg) {
-            std::vector<int> sec, seg;
-            std::vector<double> lfp_factors;
-
-            sec.reserve(nseg);
-            seg.reserve(nseg);
-            lfp_factors.reserve(total_lfp_factors);
+            auto sec = read_vector<int>(nseg);
+            auto seg = read_vector<int>(nseg);
 
-            read_array<int>(&sec[0], nseg);
-            read_array<int>(&seg[0], nseg);
+            std::vector<double> lfp_factors;
             if (total_lfp_factors > 0) {
+                // ASan reports container overflow on read_array with vec.reserve, resize does work
+                lfp_factors.resize(nseg);
```

Comment:
Just out of curiosity: did you see any discussion or threads mentioning such an issue in Asan? Sounds a bit confusing, right?